### ISSUE TITLE
IGitHubClient.Release -> IRepositoriesClient.Release

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#1032]
This is currently defined as a top-level property, but is documented under `/repos/releases` in the API:

https://developer.github.com/v3/repos/releases/

```csharp
var releases = await github.Releases.GetAll("octokit", "octokit.net");
```

We should mark the top-level property as obsolete and move this under `IRepositoriesClient`:

```csharp
var releases = await github.Repository.Releases.GetAll("octokit", "octokit.net");
```

**Comment by RobPethick** [comment]
I'll work on this one

**Comment by ryangribble** [comment]
Just going through open issues and noticed this one was fixed in #1058 and should be closed

**Comment by shiftkey** [comment]
Whoops! The auto-closing of PRs didn't catch this one. Done!
